herd/machAction.ml

```diff
@@ -96,12 +96,12 @@ end = struct
   let access_of_constant cst =
     let open Constant in
     match cst with
-    | Symbolic (Virtual _) -> VIR
-    | Symbolic (Physical _) -> PHY
+    | Symbolic (Virtual _) -> Access.VIR
+    | Symbolic (Physical _) -> Access.PHY
+    | Symbolic (TagAddr _) -> Access.TAG
```

Comment:
I don't fully understand why we have the Access field/type. But now that a TagAddr can be Virtual or Physical can we also get rid of Access.TAG?

Reply:
The Access tag is used to restrict rf matching when addresses are not known. Matching of writes and reads is performed for every compatible writes and read, compatibility being defined as having the same access field (more or less) : https://github.com/herd/herdtools7/blob/0b1fb5931d417f1fb326ff7f9adba665d5677947/herd/mem.ml#L854

Reply:
Ok and since we don't want to mix accesses by LDR/STR and other regular memory instructions with LDG/STG and implicit tag reads we need this. I see thanks!